test/kitchen/site-cookbooks/dd-security-agent-check/recipes/default.rb

```diff
@@ -5,229 +5,12 @@
-    end
-
-    service 'docker' do
-      action [ :enable, :start ]
-    end
-  elsif ['ubuntu'].include?(node[:platform])
-    docker_installation_package 'default' do
-      action :create
-      setup_docker_repo false
-      package_name 'docker.io'
-    end
-  else
-    docker_service 'default' do
-      action [:create, :start]
-    end
-  end
-
-  if node[:cws_platform] == "docker"
-    # Please see https://github.com/paulcacheux/cws-buildimages/blob/main/Dockerfile
-    # for the definition of this base image.
-    # If this successfully helps in reducing the amount of rate limits, this should be moved
-    # to DataDog/datadog-agent-buildimages.
-    file "#{wrk_dir}/Dockerfile" do
-      content <<-EOF
-      FROM ghcr.io/paulcacheux/cws-centos7@sha256:4fc1aac178b5c1690ce71c37f22b8a23cedfb969c7056702c21be50e848e554f
-
-      ADD nikos.tar.gz /opt/datadog-agent/embedded/nikos/embedded/
-
-      COPY clang-bpf /opt/datadog-agent/embedded/bin/
-      COPY llc-bpf /opt/datadog-agent/embedded/bin/
-
-      CMD sleep 7200
-      EOF
-      action :create
-    end
-
-    docker_image 'testsuite-img' do
-      tag 'latest'
-      source wrk_dir
-      action :build
-    end
-
-    docker_container 'docker-testsuite' do
-      repo 'testsuite-img'
-      tag 'latest'
-      cap_add ['SYS_ADMIN', 'SYS_RESOURCE', 'SYS_PTRACE', 'NET_ADMIN', 'IPC_LOCK', 'ALL']
-      volumes [
-        # security-agent misc
-        '/tmp/security-agent:/tmp/security-agent',
-        # HOST_* paths
-        '/proc:/host/proc',
-        '/etc:/host/etc',
-        '/sys:/host/sys',
-        # os-release
-        '/etc/os-release:/host/etc/os-release',
-        '/usr/lib/os-release:/host/usr/lib/os-release',
-        # passwd and groups
-        '/etc/passwd:/etc/passwd',
-        '/etc/group:/etc/group',
-      ]
-      env [
-        'HOST_PROC=/host/proc',
-        'HOST_ETC=/host/etc',
-        'HOST_SYS=/host/sys',
-      ]
-      privileged true
-    end
-
-    docker_exec 'debug_fs' do
-      container 'docker-testsuite'
-      command ['mount', '-t', 'debugfs', 'none', '/sys/kernel/debug']
-    end
-
-    for i in 0..7 do
-      docker_exec 'create_loop' do
-        container 'docker-testsuite'
-        command ['bash', '-c', "mknod /dev/loop#{i} b 7 #{i} || true"]
-      end
-    end
-  end
-end
-
-if platform_family?('centos', 'fedora', 'rhel')
-  selinux_state "SELinux Permissive" do
-    action :permissive
-  end
-end
-
-if File.exists?('/sys/kernel/security/lockdown')
-  file '/sys/kernel/security/lockdown' do
-    action :create_if_missing
-    content "integrity"
-  end
-end
-
-# system-probe common
-
-system_probe_tests_folder = '/tmp/system-probe-tests'
-
-directory system_probe_tests_folder do
-  recursive true
-end
+include_recipe "::functional-tests"
 
-file "#{system_probe_tests_folder}/color_idx" do
-  content node[:color_idx].to_s
-  mode 644
+if node['platform_family'] != 'windows'
+  include_recipe "::stress-tests"
 end
```

Comment:
If I understand correctly, the run lists for the security-agent tests are:
- `security-agent-test`: `["dd-agent-rhel-workaround", "dd-agent-sles-workaround", "dd-security-agent-check"]`. No specific recipes are specified, so the `default.rb` file is used for each cookbook
- `security-agent-stress`: `["dd-agent-rhel-workaround", "dd-agent-sles-workaround", "dd-security-agent-check::stress-tests"]`

Before that change, `security-agent-test` ran `default.rb` only, and `security-agent-stress` ran `stress-tests.rb` only.

Now:
- `security-agent-test` runs `default.rb`, which includes `functional-tests.rb` (which contains the previous contents of `default.rb`), _and also_ `stress-tests.rb` on Linux, which is a change in behavior I didn't expect (I'd have thought `stress-tests.rb` would only be run for the `security-agent-stress` suite).
- `security-agent-stress` still only runs `stress-tests.rb`.

Was this done on purpose?

Reply:
This was a mistake. I updated the run list and recipes to have default.rb be the place for common code, and to have the run lists include both the default.rb and the respective *-tests.rb.